memoizable.gemspec

```diff
@@ -17,6 +17,4 @@ Gem::Specification.new do |gem|
   gem.extra_rdoc_files = Dir['**/*.md']
 
   gem.add_runtime_dependency('thread_safe', '~> 0.3', '>= 0.3.4')
-
-  gem.add_development_dependency('bundler', '~> 1.7', '>= 1.7.9')
```

Comment:
@dkubb I'd normally extract this to a separate commit. But as the build has bitrotted (we last touched this repo in 2014!) I have to do this in one step.

Rationale for removing it: Explicit bundler dependency is not required these days. And even if we added one given the low commit frequency of this repo we'll end up in having to bump it always.